Xamarin.Forms.Core/MessagingCenter.cs

```diff
@@ -2,12 +2,30 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
-using System.Runtime.CompilerServices;
 
 namespace Xamarin.Forms
 {
-	public static class MessagingCenter
+	public interface IMessagingCenter
 	{
+		void Send<TSender, TArgs>(TSender sender, string message, TArgs args) where TSender : class;
+
+		void Send<TSender>(TSender sender, string message) where TSender : class;
+
+		void Subscribe<TSender, TArgs>(object subscriber, string message, Action<TSender, TArgs> callback, TSender source = null) where TSender : class;
+
+		void Subscribe<TSender>(object subscriber, string message, Action<TSender> callback, TSender source = null) where TSender : class;
+
+		void Unsubscribe<TSender, TArgs>(object subscriber, string message) where TSender : class;
+
+		void Unsubscribe<TSender>(object subscriber, string message) where TSender : class;
+	}
+
+	public class MessagingCenter : IMessagingCenter
+	{
+		static readonly Lazy<MessagingCenter> s_instance = new Lazy<MessagingCenter>(() => new MessagingCenter());
+
+		public static IMessagingCenter Instance => s_instance.Value;
```

Comment:
what's the point of the `Lazy` ? Allocations ? I don't think we gain anything measurable by instantiating a `Lazy<MC>` (and the Func) instead of a `MessagingCenter` in `.cctor` (static constructor), and at the end of the day, we have allocated one object more...